pkg/services/notifications/mailer.go

```diff
@@ -35,11 +38,14 @@ func (ns *NotificationService) send(msg *Message) (int, error) {
 		m.SetBody("text/html", msg.Body)
 
 		if err := dialer.DialAndSend(m); err != nil {
-			return 0, err
+			werr = errors.Wrapf(err, "Failed to send notification to email address: %s", address)
+			continue
 		}
+
+		num += 1
 	}
 
-	return len(msg.To), nil
+	return num, werr
```

Comment:
since the loop never is exicted `num` will always be the same as `len(msg.To)` So unless there is somhing I havent consiered you can just go back to using `len(msg.To)`

Reply:
https://github.com/grafana/grafana/blob/0c1530c7a89ba3321a24567a28a511cd13a6c407/pkg/services/notifications/notifications.go#L85-L96

the `num` should be the number of addresses sent successfully.

If some addresses failed, `num` will less than `len(msg.To)`.

Reply:
Your right. Silly me :)